.github/workflows/snapshot-file-change.yml

```diff
@@ -71,11 +71,14 @@ jobs:
           destination: '${{ inputs.destination_prefix }}/pmap-file-change-${{ github.run_id }}-${{ github.run_attempt }}'
           headers: |-
             content-type: application/json
-            x-goog-meta-git-commit: ${{ github.sha }}
-            x-goog-meta-git-repo: ${{ github.repository }}
-            x-goog-meta-git-workflow: ${{ github.workflow }}
-            x-goog-meta-git-workflow-sha: ${{ github.workflow_sha }}
-            x-goog-meta-git-workflow-triggered-timestamp: ${{ env.CUR_TIMESTAMP }}
+            x-goog-meta-github-commit: ${{ github.sha }}
+            x-goog-meta-github-repo: ${{ github.repository }}
+            x-goog-meta-github-run-attempt: ${{github.run_attempt}}
+            x-goog-meta-github-run-id: ${{github.run_id}}
+            x-goog-meta-github-workflow: ${{ github.workflow }}
+            x-goog-meta-github-workflow-sha: ${{ github.workflow_sha }}
+            x-goog-meta-github-workflow-triggered-timestamp: ${{ env.CUR_TIMESTAMP }}
+            x-goog-meta-gcs-file-path: ${{ inputs.destination_prefix }}/pmap-file-change-${{ github.run_id }}-${{ github.run_attempt }}/${{inputs.path}}
```

Comment:
You have an misunderstanding about the requirement. I think this should record the file path of the file in the github repo.

Reply:
The point of recording the github filepath, github repo, github commit to that we can directly find the data in the github which maps in the data in BQ.

Reply:
Now for example, a file  `folder1/folder2/aaa.yml` will be uploaded to GCS in `${{ inputs.destination_prefix }}/pmap-file-change-${{ github.run_id }}-${{ github.run_attempt }}/folder1/folder2/aaa.yml`. And the we will parse that and `FilePath` in `GithubSource` as `folder1/folder2/aaa.yml`